src/packages/version-check/tests/version-check.test.ts

```diff
@@ -0,0 +1,737 @@
+
+import { VersionCheckOptions } from "../src/types";
+import { VersionCheck } from "../src/version-check";
+import { semverUpgradeType } from "../src/semver";
+import http2 from "http2";
+import assert from "assert";
+import * as fs from "fs";
+
+describe("@ganache/version-check", () => {
+  let vc;
+  const testVersion = "0.0.0";
+  const versionString = "v1.2.3";
+  const version = "1.2.3";
+
+  const testConfig: VersionCheckOptions = {
+    packageName: "test",
+    enabled: true,
+    lastNotification: 0,
+    url: "test",
+    ttl: 100,
+    latestVersion: "99.99.99",
+    lastVersionLogged: "99.99.90",
+    disableInCI: false,
+    activated: false
+  };
+  const sparseConfig: VersionCheckOptions = {
+    packageName: "test",
+    enabled: true,
+    url: "test",
+    ttl: 100,
+    latestVersion: "99.99.99",
+    disableInCI: false
+  };
+
+  let message: string = "";
+  const testLogger = { log: str => (message = str) };
+
+  beforeEach(() => {
+    process.env.IGNORE_ISCI = "true";
+    vc = new VersionCheck(testVersion);
+
+    message = "";
+  });
+
+  afterEach(() => {
+    const testConfigFileLocation = vc.configFileLocation;
+    if (fs.existsSync(testConfigFileLocation)) {
+      fs.unlinkSync(testConfigFileLocation);
+    }
+    delete process.env.IGNORE_ISCI;
+  });
+
+  describe("constructor", () => {
+    it("sets the current version", () => {
+      assert.strictEqual(
+        vc._currentVersion,
+        testVersion,
+        "currentVersion incorrectly set by constructor."
+      );
+    });
+    it("instantiates with the default config", () => {
+      const config = VersionCheck.DEFAULTS;
+
+      assert.deepStrictEqual(
+        vc._config,
+        config,
+        "Default Config values do not match newly created version checker."
+      );
+    });
+    it("sets an optional config", () => {
+      vc = new VersionCheck(testVersion, testConfig);
+
+      assert.deepStrictEqual(
+        vc._config,
+        testConfig,
+        "testConfig is not set in the constructor."
+      );
+    });
+    it("sets an optional sparse config", () => {
+      const expectedConfig = {
+        ...VersionCheck.DEFAULTS,
+        ...sparseConfig
+      };
+
+      vc = new VersionCheck(testVersion, sparseConfig);
+
+      assert.deepStrictEqual(
+        vc._config,
+        expectedConfig,
+        "Sparse config values incorrectly set in constructor."
+      );
+    });
+
+    it("strips invalid config properties from the constructor param", () => {
+      const bogusConfig = {
+        iShouldNotExist: "Existential Panic",
+        ...testConfig
+      };
+      vc = new VersionCheck(testVersion, bogusConfig);
+
+      assert.deepStrictEqual(
+        vc._config,
+        testConfig,
+        "Invalid config properties are not removed from config."
+      );
+    });
+    it("disables if currentVersion is not a valid semver.", () => {
+      delete process.env.IGNORE_ISCI;
+      vc = new VersionCheck("");
```

Comment:
```suggestion
      vc = new VersionCheck("invalid-version");
```
is just a little bit clearer

Reply:
I don't really want to split hairs on magic values, is there a functional reason to change this or is it just preference that you'd like to see something there?

Reply:
I know it's a little bit bike-shedding, but I think literals that are used in tests should be as self-describing as possible